controller/controller_server/utils.py

```diff
@@ -359,3 +422,32 @@ def get_current_timestamp():
 
 def hash_string(string):
     return base58.b58encode(sha256(string.encode()).digest()).decode()
+
+
+def validate_parameters_match_volume(parameters, volume):
```

Comment:
looks to me like there is some code duplication in this function.
consider using a common generic function, e.g.:
```
from operator import eq

def _validate_parameter_match_volume(parameters, parameter_name, value_from_volume, error_message_format, cmp=eq):
    parameter_value = parameters.get(parameter_name)
    if parameter_value:
        if not cmp(parameter_value, value_from_volume):
            raise ValidationException(error_message_format.format(parameter_value, value_from_volume))
```

Reply:
what about `volume_name.startswith` for `prefix`?

Reply:
```
_validate_parameter_match_volume(parameters, config.PARAMETERS_VOLUME_NAME_PREFIX,
                                 volume.name, messages.prefix_not_match_volume_message,
                                 lambda name, prefix: name.startswith(prefix + NAME_PREFIX_SEPARATOR))
```

Reply:
nevermind, it wouldn't fit with the space efficiency `else` part

Reply:
there is else for one of them and its getting to complicated